### PR TITLE
For upstream: rework

### PR DESCRIPTION
**Pull request by c0d3z3r0**
Some rework changes - maybe you like it ;)

### CODE REVIEW

File: MEA.py [hunk 1]
```diff
@@ -12686,15 +12677,15 @@ def mass_scan(f_path) :
...
+				effs_match = (re.compile(br'EFFSOSID')).search(reading) # EFFSOSID detection
 				if effs_match is not None :
 					(start_effs_match, end_effs_match) = effs_match.span()
 					effs_start = int.from_bytes(reading[end_effs_match:end_effs_match + 0x4], 'little')
 					effs_size = int.from_bytes(reading[end_effs_match + 0x4:end_effs_match + 0x8], 'little')
 					effs_data = reading[fpt_start + effs_start:fpt_start + effs_start + effs_size]
 					
-					me3_type_fix1 = (re.compile(br'\x4D\x45\x5F\x43\x46\x47\x5F\x44\x45\x46\x04\x4E\x56\x4B\x52')).findall(effs_data) # ME_CFG_DEF.NVKR detection (RGN have <= 2)
-					me3_type_fix2 = (re.compile(br'\x4D\x61\x78\x55\x73\x65\x64\x4B\x65\x72\x4D\x65\x6D\x04\x4E\x56\x4B\x52\x7F\x78\x01')).search(effs_data) # MaxUsedKerMem.NVKR.x. detection
+					me3_type_fix1 = (re.compile(br'ME_CFG_DEF.NVKR')).findall(effs_data) # ME_CFG_DEF.NVKR detection (RGN have <= 2)
```
Comment (platomav) [inline MEA.py:12687]: The dot should be \x04 instead

File: MEA.py [hunk 2]
```diff
@@ -12776,16 +12767,16 @@ def mass_scan(f_path) :
...
 					effs_start = int.from_bytes(reading[end_effs_match:end_effs_match + 0x4], 'little')
 					effs_size = int.from_bytes(reading[end_effs_match + 0x4:end_effs_match + 0x8], 'little')
 					effs_data = reading[fpt_start + effs_start:fpt_start + effs_start + effs_size]
 				
-					me4_type_fix1 = (re.compile(br'\x4D\x45\x5F\x43\x46\x47\x5F\x44\x45\x46')).findall(effs_data) # ME_CFG_DEF detection (RGN have 2-4)
-					me4_type_fix2 = (re.compile(br'\x47\x50\x49\x4F\x31\x30\x4F\x77\x6E\x65\x72')).search(effs_data) # GPIO10Owner detection
-					me4_type_fix3 = (re.compile(br'\x41\x70\x70\x52\x75\x6C\x65\x2E\x30\x33\x2E\x30\x30\x30\x30\x30\x30')).search(effs_data) # AppRule.03.000000 detection
+					me4_type_fix1 = (re.compile(br'ME_CFG_DEF')).findall(effs_data) # ME_CFG_DEF detection (RGN have 2-4)
+					me4_type_fix2 = (re.compile(br'GPIO10Owner')).search(effs_data) # GPIO10Owner detection
+					me4_type_fix3 = (re.compile(br'AppRule.03.000000')).search(effs_data) # AppRule.03.000000 detection
```
Comment (platomav) [inline MEA.py:12779]: An escape for each dot character is required here, otherwise they match everything except new line

**Review by platomav — CHANGES_REQUESTED** [review]
Thank you for the rework. I've left comments to some regex patterns which need to be fixed first.

File: MEA.py [hunk 3]
```diff
@@ -12686,15 +12677,15 @@ def mass_scan(f_path) :
...
 				if effs_match is not None :
 					(start_effs_match, end_effs_match) = effs_match.span()
 					effs_start = int.from_bytes(reading[end_effs_match:end_effs_match + 0x4], 'little')
 					effs_size = int.from_bytes(reading[end_effs_match + 0x4:end_effs_match + 0x8], 'little')
 					effs_data = reading[fpt_start + effs_start:fpt_start + effs_start + effs_size]
 					
-					me3_type_fix1 = (re.compile(br'\x4D\x45\x5F\x43\x46\x47\x5F\x44\x45\x46\x04\x4E\x56\x4B\x52')).findall(effs_data) # ME_CFG_DEF.NVKR detection (RGN have <= 2)
-					me3_type_fix2 = (re.compile(br'\x4D\x61\x78\x55\x73\x65\x64\x4B\x65\x72\x4D\x65\x6D\x04\x4E\x56\x4B\x52\x7F\x78\x01')).search(effs_data) # MaxUsedKerMem.NVKR.x. detection
+					me3_type_fix1 = (re.compile(br'ME_CFG_DEF.NVKR')).findall(effs_data) # ME_CFG_DEF.NVKR detection (RGN have <= 2)
+					me3_type_fix2 = (re.compile(br'MaxUsedKerMem\x04NVKR\x7F\x78\x01')).search(effs_data) # MaxUsedKerMem.NVKR.x. detection
```
Comment (c0d3z3r0) [inline MEA.py:12688]: \x78 -> x

File: MEA.py (hunk 1 above)
Comment (c0d3z3r0) [reply MEA.py:12687]: oops, good catch!

File: MEA.py (hunk 2 above)
Comment (c0d3z3r0) [reply MEA.py:12779]: ack

File: MEA.py [hunk 4]
```diff
@@ -12624,9 +12615,9 @@ def mass_scan(f_path) :
 							me2_type_fix = int.from_bytes(reading[qstpat_start:qstpat_end], 'big')
 							me2_type_exp = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
 				elif sku == 'AMT' and minor < 5 :
-					nvsh_match = (re.compile(br'\x4E\x56\x53\x48\x4F\x53\x49\x44')).search(reading) # NVSHOSID detection
+					nvsh_match = (re.compile(br'NVSHOSID')).search(reading) # NVSHOSID detection
 					if nvsh_match is not None :
-						netip_match = (re.compile(br'\x6E\x65\x74\x2E\x69\x70\xFF\xFF\xFF')).search(reading) # "net.ip" detection (2.0-2.2)
+						netip_match = (re.compile(br'net.ip\xFF\xFF\xFF')).search(reading) # "net.ip" detection (2.0-2.2)
```
[the remark this answers is not in the document]
Comment (c0d3z3r0) [reply MEA.py:12620]: ack

File: MEA.py (hunk 3 above)
Comment (c0d3z3r0) [reply MEA.py]: done

**Review by platomav — APPROVED** [review]

**Pull request merged**